test/stableTRIStaking/transfer.test.ts

```diff
@@ -0,0 +1,313 @@
+    expect(await this.pTRI.balanceOf(this.deployer.address)).to.equal(0);
+
+    expect(await this.rewardToken.balanceOf(this.deployer.address)).to.equal(depositAmount);
+    expect(await this.rewardToken.balanceOf(this.pTRI.address)).to.equal(0);
+    expect(await this.pTRI.balanceOf(this.deployer.address)).to.equal(0);
+    expect(await this.pTRI.balanceOf(this.escrow.address)).to.equal(depositAmount);
+
+    let [deployerAmount, deployerRewardDebt] = await this.pTRI
+      .connect(this.deployer)
+      .getUserInfo(this.deployer.address, this.rewardToken.address);
+
+    expect(deployerAmount).to.equal(0);
+    expect(deployerRewardDebt).to.equal(0);
+
+    let [escrowAmount, escrowRewardDebt] = await this.pTRI.getUserInfo(this.escrow.address, this.rewardToken.address);
+
+    expect(escrowAmount).to.equal(100);
+    expect(escrowRewardDebt).to.equal(100);
+
+    await this.rewardToken.transfer(this.pTRI.address, depositAmount);
+    expect(await this.rewardToken.balanceOf(this.deployer.address)).to.equal(0);
+    expect(await this.pTRI.pendingReward(this.deployer.address, this.rewardToken.address)).to.equal(0);
+    expect(await this.pTRI.pendingReward(this.escrow.address, this.rewardToken.address)).to.equal(100);
+
+    await this.escrow.connect(this.deployer).withdraw(this.deployer.address, this.pTRI.address, depositAmount);
+
+    [escrowAmount, escrowRewardDebt] = await this.pTRI.getUserInfo(this.escrow.address, this.rewardToken.address);
+
+    expect(escrowAmount).to.equal(0);
+    expect(escrowRewardDebt).to.equal(0);
+
+    expect(await this.pTRI.balanceOf(this.deployer.address)).to.equal(100);
+    expect(await this.pTRI.balanceOf(this.escrow.address)).to.equal(0);
+
+    [escrowAmount, escrowRewardDebt] = await this.pTRI.getUserInfo(this.escrow.address, this.rewardToken.address);
+
+    expect(escrowAmount).to.equal(0);
+    expect(escrowRewardDebt).to.equal(0);
+
+    [deployerAmount, deployerRewardDebt] = await this.pTRI
+      .connect(this.deployer)
+      .getUserInfo(this.deployer.address, this.rewardToken.address);
+
+    expect(deployerAmount).to.equal(100);
+    expect(deployerRewardDebt).to.equal(200);
+
+    expect(await this.rewardToken.balanceOf(this.deployer.address)).to.equal(0);
+    expect(await this.rewardToken.balanceOf(this.escrow.address)).to.equal(100);
+
+    await this.pTRI.connect(this.deployer).withdraw(depositAmount);
+    expect(await this.pTRI.balanceOf(this.deployer.address)).to.equal(0);
+    expect(await this.tri.balanceOf(this.deployer.address)).to.equal(100);
+    expect(await this.rewardToken.balanceOf(this.escrow.address)).to.equal(100);
+
+    expect(await this.pTRI.pendingReward(this.deployer.address, this.rewardToken.address)).to.equal(0);
+    expect(await this.pTRI.pendingReward(this.escrow.address, this.rewardToken.address)).to.equal(0);
+  });
+
+  it("should claim before deposit and withdraw from escrow, escrow claims on withdrawal then user withdraws from pTRI", async function () {
```

Comment:
too big a test can you add comments after every transaction?
no need to add comments after every state check, only after every transaction